flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/output/BootstrapStreamTask.java

```diff
@@ -82,16 +86,14 @@ protected void init() throws Exception {
         mainOperator = mainOperatorAndTimeService.f0;
         mainOperator.initializeState(createStreamTaskStateInitializer());
         mainOperator.open();
+        recordProcessor = RecordProcessorUtils.getRecordProcessor(mainOperator);
     }
 
     @Override
     protected void processInput(MailboxDefaultAction.Controller controller) throws Exception {
         StreamElement element = input.take();
         if (element.isRecord()) {
-            StreamRecord<IN> streamRecord = element.asRecord();
-
-            mainOperator.setKeyContextElement1(streamRecord);
-            mainOperator.processElement(streamRecord);
+            recordProcessor.accept(element.asRecord());
```

Comment:
Do you think we need to add `hasKeyContext1()` and `hasKeyContext2()` to  `StateBootstrapWrapperOperator`?

Reply:
I think it's not needed because all invocations to operators are intercepted. Operators' responsibility is to provide correct information of `hasKeyContext`.

Reply:
Sounds good. Thanks for the explanation.